### DISCUSSION TITLE
otlp http exporter no need to import grpc

### DISCUSSION THREAD

**Discussion by wxl374** [open-telemetry/opentelemetry-go#2509] [Q&A]
I add traces in my project which imported an old version grpc, but I got compile error for lack of google.golang.org/grpc/backoff (imported by [options.go](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/internal/otlpconfig/options.go)) directory. I tried to upgrade grpc, but there are some compatible issues with my project (we implemented our own balancer because of this [grpc issue](https://github.com/grpc/grpc-go/issues/2191)). There is no need to import grpc when we use http exporter, maybe we should separate http exporter and grpc exporter in different directories. Any ideas to solve this problem?

**Comment by dmathieu** [comment]
The [otlptrace](https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlptrace) package has two subpackages, one for HTTP and the other for GRPC.
Though this used not to be the case, so maybe you're still using an old version of the library?

Do you have a code sample showing how you're using the exporter?

**Reply by wxl374** [reply]
I use version [v1.3.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.3.0)
I just copy code from https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/otlptracehttp/example_test.go, and invoke https://github.com/open-telemetry/opentelemetry-go/blob/73edf3dfcca4b0696597b0412ba3b5106ed794e7/exporters/otlp/otlptrace/otlptracehttp/example_test.go#L67 (modify the prefix "i" to "I") in my project.
